lib/shopping_cart.rb

```diff
@@ -0,0 +1,28 @@
+class ShoppingCart
+  attr_reader :name,
+              :capacity,
+              :products
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    cart_details = { name: name, capacity: capacity }
+  end
+
+  def total_number_of_products
+    cart_products.sum
```

Comment:
This the right enumerable! Just keep working at enumerables to nail down the implementation